pkg/server/api/svid/v1/service_test.go

```diff
@@ -921,10 +926,206 @@ func TestServiceBatchNewX509SVID(t *testing.T) {
+		SpiffeId:   &types.SPIFFEID{TrustDomain: "example.org", Path: ""},
+		Downstream: false,
+	}
+
+	downstreamEntryWrongTd := &types.Entry{
+		Id:         "downstreamCA2",
+		ParentId:   api.ProtoFromID(agentID),
+		SpiffeId:   &types.SPIFFEID{TrustDomain: "example-2.org", Path: ""},
+		Downstream: false,
+	}
+
+	x509CA := test.ca.X509CA()
+	test.ef.entries = []*types.Entry{downstreamEntry, downstreamEntryUnauthorized}
+
+	for _, tt := range []downstreamCaTest{
+		{
+			name:           "Malformed CSR",
+			rateLimiterErr: nil,
+			err:            "malformed CSR: asn1: structure error",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.InvalidArgument,
+			fetcherErr:     "",
+			entry:          downstreamEntry,
+		},
+		{
+			name:           "Rate Limiter Err",
+			rateLimiterErr: status.Error(codes.Internal, "rate limit error"),
+			err:            "rate limit error",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.Internal,
+			fetcherErr:     "",
+			entry:          downstreamEntry,
+		},
+		{
+			name:           "Unauthorized",
+			rateLimiterErr: nil,
+			err:            "error not found or not authorized",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.NotFound,
+			fetcherErr:     "",
+			entry:          downstreamEntryUnauthorized,
+		},
+		{
+			name:           "Nonexistent entry",
+			rateLimiterErr: nil,
+			err:            "error not found or not authorized",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.NotFound,
+			fetcherErr:     "",
+			entry:          &types.Entry{Id: "non-existent-entry"},
+		},
+		{
+			name:           "Entry fetch error",
+			rateLimiterErr: nil,
+			err:            "failed to fetch registration entries",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.Internal,
+			fetcherErr:     "Entry fetch error",
+			entry:          downstreamEntry,
+		},
+		{
+			name:           "FailDataStore",
+			rateLimiterErr: nil,
+			err:            "bundle not found",
+			failSigning:    false,
+			csrTemplate:    &x509.CertificateRequest{},
+			code:           codes.Internal,
+			fetcherErr:     "",
+			entry:          downstreamEntry,
+			failDataStore:  true,
+		},
+		{
+			name:           "CertRequestWrongTd",
+			rateLimiterErr: nil,
+			err:            "error not found or not authorized",
+			failSigning:    false,
+			failDataStore:  false,
+			csrTemplate: &x509.CertificateRequest{
+				URIs: []*url.URL{workloadID.URL()},
+			},
+			fetcherErr: "",
+			entry:      downstreamEntryWrongTd,
+			code:       codes.NotFound,
+		},
+		{
+			name:           "SuccessfulCARequest",
+			rateLimiterErr: nil,
+			err:            "",
+			failSigning:    false,
+			failDataStore:  false,
+			csrTemplate: &x509.CertificateRequest{
+				// Right now, the request will succeed even if URIs is empty. This is
```

Comment:
From the API documentation:
```
    // Required. The ASN.1 DER encoded Certificate Signing Request (CSR). The
    // CSR is only used to convey the public key; other fields in the CSR are
    // ignored. The X509-SVID attributes are determined by the entry.
 ```

The CSR is only used to convey the public key. The SPIFFE ID of the returned CA certificate should be the trust domain ID (e.g. `spiffe://example.org`). This is a small departure in behavior from the current node FetchX509CASVID RPC.